aea/helpers/dialogue/base.py

```diff
@@ -391,20 +386,20 @@ def _basic_rules(self, message: Message) -> bool:
         :return: True if valid, False otherwise.
         """
         message_id = message.message_id
-        target = cast(int, message.get("target"))
-        performative = cast(Enum, message.get("performative"))
+        target = message.target
+        performative = message.performative
 
-        if self.is_empty:
+        if self.last_message is None:
```

Comment:
why did you change this?

Reply:
Because the implementation meant one had to then check again if the last message was present for pypi. It makes more sense this way.